tests/plugins/vllm_add_dummy_platform/vllm_add_dummy_platform/dummy_custom_ops.py

```diff
@@ -0,0 +1,26 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+from typing import Optional
+
+import torch
+
+from vllm.model_executor.custom_op import CustomOp
+from vllm.model_executor.layers.rotary_embedding import RotaryEmbedding
+
+
+@CustomOp.register("RotaryEmbedding", is_oot_custom_op=True)
+class DummyRotaryEmbedding(RotaryEmbedding):
+    """Original rotary positional embedding."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.addition_config = True
+
+    def forward_oot(
+        self,
+        positions: torch.Tensor,
+        query: torch.Tensor,
+        key: torch.Tensor,
+        offsets: Optional[torch.Tensor] = None,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        return super().forward_oot(positions, query, key, offsets)
```

Comment:
What's the semantics of this? Does this completely replace the pre-existing RotaryEmbedding custom op?

Also, under what context does forward_oot get called?

Reply:
foward_oot is called based on existing custom_op forward_method strategy, which will be if the platform is registered as OOT platform, it will call forward_oot.
And, yes, this customRotaryEmbedding class aimed to replace the existing in-tree rotary_embedding class. Since the class is derive from in-tree, we can still use most of existing strategy but also flexible to add additional steps.
For example, to HPU, we will need to override process_after_model_loading to help do weight process for HPU. We will also need to override any complex data_type preparation due to HPU is not supporting complex yet.